src/main/java/com/google/firebase/remoteconfig/internal/RemoteConfigServiceErrorResponse.java

```diff
@@ -60,9 +54,9 @@ public RemoteConfigErrorCode getRemoteConfigErrorCode() {
       return null;
     }
 
-    int separator = message.indexOf(':');
-    if (separator != -1) {
-      String errorCode = message.substring(0, separator).replaceAll("\\[|\\]", "");
+    Matcher errorMatcher = Pattern.compile("^\\[(\\w+)\\]:.*$").matcher(message);
```

Comment:
`Pattern.compile()` could be expensive. Pre-compute it and store as a static constant.

Reply:
Good point! Thanks!